core/protocols/aws-json-protocol/src/main/java/software/amazon/awssdk/protocols/json/BaseAwsJsonProtocolFactory.java

```diff
@@ -138,14 +149,15 @@ private <T> MetricCollectingHttpResponseHandler<T> timeUnmarshalling(HttpRespons
     }
 
     private StructuredJsonGenerator createGenerator(OperationInfo operationInfo) {
-        AwsJsonProtocol protocol = protocolMetadata.protocol();
-        if (operationInfo.hasPayloadMembers()
-            || protocol == AwsJsonProtocol.AWS_JSON
-            || protocol == AwsJsonProtocol.SMITHY_RPC_V2_CBOR) {
-            return createGenerator();
-        } else {
+        Boolean useNoOp = operationInfo.addtionalMetadata(USE_NO_OP_GENERATOR);
+        if (useNoOp == null) {
+            AwsJsonProtocol protocol = protocolMetadata.protocol();
+            useNoOp = !(operationInfo.hasPayloadMembers() || protocol == AwsJsonProtocol.AWS_JSON);
```

Comment:
I find this hard to read and had to stare at it for a while. How about using DeMorgan's law and change to 
`!operationInfo.hasPayloadMembers() && !(protocol == AwsJsonProtocol.AWS_JSON))`? I think that's more intuitive.

Reply:
Sure, I can change. What about introducing the negation also?

```
!operationInfo.hasPayloadMembers() && protocol != AwsJsonProtocol.AWS_JSON
```

Reply:
yeah!